src/Aliases.jl

```diff
@@ -1,84 +1,84 @@
+Base.@deprecate_binding iszero_row is_zero_row
+Base.@deprecate_binding Localization localization
 
 #
-# Some deprecation aliases follow. They use @alias instead of @deprecate
+# Some deprecation aliases follow. They use Base.@deprecate_binding instead of @deprecate
 # because there are still packages adding methods for them, which would break
 # if we used @deprecate. Once those packages are updated, we can switch to
 # using @deprecate
 #
 
 # Deprecated in 0.27.*
 
-@alias MatrixSpace matrix_space
+Base.@deprecate_binding MatrixSpace matrix_space
 
-@alias PolynomialRing polynomial_ring
-@alias PowerSeriesRing power_series_ring
-@alias LaurentPolynomialRing laurent_polynomial_ring
-@alias LaurentSeriesRing laurent_series_ring
-@alias LaurentSeriesField laurent_series_field
-@alias FractionField fraction_field
-@alias NumberField number_field
+Base.@deprecate_binding PolynomialRing polynomial_ring
+Base.@deprecate_binding PowerSeriesRing power_series_ring
+Base.@deprecate_binding LaurentPolynomialRing laurent_polynomial_ring
+Base.@deprecate_binding LaurentSeriesRing laurent_series_ring
+Base.@deprecate_binding LaurentSeriesField laurent_series_field
+Base.@deprecate_binding FractionField fraction_field
+Base.@deprecate_binding NumberField number_field
 
-@alias FunctionField function_field
-@alias RationalFunctionField rational_function_field
+Base.@deprecate_binding FunctionField function_field
+Base.@deprecate_binding RationalFunctionField rational_function_field
 
-@alias PolyElem PolyRingElem
-@alias NCPolyElem NCPolyRingElem
-@alias MPolyElem MPolyRingElem
-@alias UnivPolyElem UniversalPolyRingElem
-@alias LaurentPolyElem LaurentPolyRingElem
-@alias LaurentMPolyElem LaurentMPolyRingElem
+Base.@deprecate_binding PolyElem PolyRingElem
+Base.@deprecate_binding NCPolyElem NCPolyRingElem
+Base.@deprecate_binding MPolyElem MPolyRingElem
+Base.@deprecate_binding UnivPolyElem UniversalPolyRingElem
+Base.@deprecate_binding LaurentPolyElem LaurentPolyRingElem
+Base.@deprecate_binding LaurentMPolyElem LaurentMPolyRingElem
 
-@alias UnivPolyRing UniversalPolyRing
+Base.@deprecate_binding UnivPolyRing UniversalPolyRing
 
-@alias AbsSeriesRing AbsPowerSeriesRing
-@alias AbsSeriesElem AbsPowerSeriesRingElem
-@alias RelSeriesRing RelPowerSeriesRing
-@alias RelSeriesElem RelPowerSeriesRingElem
+Base.@deprecate_binding AbsSeriesRing AbsPowerSeriesRing
+Base.@deprecate_binding AbsSeriesElem AbsPowerSeriesRingElem
+Base.@deprecate_binding RelSeriesRing RelPowerSeriesRing
+Base.@deprecate_binding RelSeriesElem RelPowerSeriesRingElem
 
-@alias FreeAssociativeAlgebra free_associative_algebra
+Base.@deprecate_binding FreeAssociativeAlgebra free_associative_algebra
 
-@alias TotalRingOfFractions total_ring_of_fractions
+Base.@deprecate_binding TotalRingOfFractions total_ring_of_fractions
 
-@alias DirectSum direct_sum
+Base.@deprecate_binding DirectSum direct_sum
 
 # Deprecated in 0.34.*
 @alias PuiseuxSeriesField puiseux_series_field
```

Comment:
I did not touch these "new" aliases because they were just introduced... but then again I think this is only used in Nemo, and we might just as well rip of the band aid now... @fieker @thofma thoughts?

Reply:
(The comment with ripping off the band aid referred only to `puiseux_series_field` and `puiseux_series_ring` (resp. their old names), i.e.: we could use `Base.@deprecate_binding` for both right now...

Reply:
Yeah, I think we should just do it.

Reply:
@fieker also agrees, will do